src/main/java/io/fabric8/maven/docker/config/CopyConfiguration.java

```diff
@@ -0,0 +1,142 @@
+package io.fabric8.maven.docker.config;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+import io.fabric8.maven.docker.util.DeepCopy;
+
+public class CopyConfiguration implements Serializable {
+
+    public static final String CONTAINER_PATH_PROPERTY = "containerPath";
+    public static final String HOST_DIRECTORY_PROPERTY = "hostDirectory";
+
+    public static class Entry implements Serializable {
+
+        /**
+         * Full path to container file or container directory which needs to copied.
+         */
+        private String containerPath;
+
+        /**
+         * Path to a host directory where the files copied from container need to be placed. If relative path is
+         * provided then project base directory is considered as a base of that relative path. Can be <code>null</code>
+         * meaning the same as empty string. Note that if containerPath points to a directory, then a directory with the
+         * same name will be created in the hostDirectory, i.e. not just content of directory is copied, but the same
+         * name directory is created too.
+         */
+        private String hostDirectory;
+
+        public Entry() {}
+
+        public Entry(String containerPath, String hostDirectory) {
+            this.containerPath = containerPath;
+            this.hostDirectory = hostDirectory;
+        }
+
+        public String getContainerPath() {
+            return containerPath;
+        }
+
+        public String getHostDirectory() {
+            return hostDirectory;
+        }
+
+        public void setContainerPath(String containerPath) {
+            this.containerPath = containerPath;
+        }
+
+        public void setHostDirectory(String hostDirectory) {
+            this.hostDirectory = hostDirectory;
+        }
+    }
+
+    /**
+     * Items to copy from container.
+     */
+    @Parameter
+    private List<Entry> entries;
+
+    public List<Entry> getEntries() {
+        return entries;
+    }
+
+    public List<Properties> getEntriesAsListOfProperties() {
+        if (entries == null) {
+            return null;
```

Comment:
```suggestion
            return Collections.emptyList();
```

Reply:
The [same answer](https://github.com/fabric8io/docker-maven-plugin/pull/1405#discussion_r599661388) is actual here. Why I personally don;t like `null` in references representing collections, it seems to be existing design which implies `null` value if some configuration part is undefined.